tests/platform_test_harness/graphics_platform_test_harness.cpp

```diff
@@ -0,0 +1,676 @@
+            }
+            mir::optional_value<std::shared_ptr<mg::SetGbmExtension>> set_gbm_extension() override
+            {
+                return {};
+            }
+            mir::optional_value<mir::Fd> drm_fd() override
+            {
+                return {};
+            }
+            auto platform_operation(unsigned int /*op*/, mir::graphics::PlatformOperationMessage const& /*request*/)
+                -> mg::PlatformOperationMessage override
+            {
+                BOOST_THROW_EXCEPTION((std::runtime_error{"No support for mirclient"}));
+            }
+        };
+
+        return std::make_shared<NullPlatformAuthentication>();
+    }
+private:
+    std::thread main_loop_thread;
+    static char const* argv[];
+};
+char const* MinimalServerEnvironment::argv[] = {"graphics_platform_test_harness", nullptr};
+
+std::string describe_probe_result(mg::PlatformPriority priority)
+{
+    if (priority == mg::PlatformPriority::unsupported)
+    {
+        return "UNSUPPORTED";
+    }
+    else if (priority == mg::PlatformPriority::dummy)
+    {
+        return "DUMMY";
+    }
+    else if (priority < mg::PlatformPriority::supported)
+    {
+        return std::string{"SUPPORTED - "} + std::to_string(mg::PlatformPriority::supported - priority);
+    }
+    else if (priority == mg::PlatformPriority::supported)
+    {
+        return "SUPPORTED";
+    }
+    else if (priority < mg::PlatformPriority::best)
+    {
+        return std::string{"SUPPORTED + "} + std::to_string(priority - mg::PlatformPriority::supported);
+    }
+    else if (priority == mg::PlatformPriority::best)
+    {
+        return "BEST";
+    }
+    return std::string{"BEST + "} + std::to_string(priority - mg::PlatformPriority::best);
+}
+
+bool test_probe(mir::SharedLibrary const& dso, MinimalServerEnvironment& config)
+{
+    try
+    {
+        auto probe_fn =
+            dso.load_function<mg::PlatformProbe>("probe_graphics_platform", MIR_SERVER_GRAPHICS_PLATFORM_VERSION);
+
+        auto const result = probe_fn(
+            config.console_services(),
+            *std::dynamic_pointer_cast<mir::options::ProgramOption>(config.options()));
+
+        std::cout << "Probe result: " << describe_probe_result(result) << "(" << result << ")" << std::endl;
+
+        return result > mg::PlatformPriority::dummy;
+    }
+    catch (std::exception const& err)
+    {
+        std::cout << "Probing failed: " << boost::diagnostic_information(err) << std::endl;
+        return false;
+    }
+    catch (...)
+    {
+        std::cout << "Probing failed with broken exception!" << std::endl;
+        return false;
+    }
+}
+
+auto test_display_platform_construction(mir::SharedLibrary const& dso, MinimalServerEnvironment& env)
+-> std::shared_ptr<mir::graphics::DisplayPlatform>
+{
+    try
+    {
+        auto create_display_platform = dso.load_function<mg::CreateDisplayPlatform>(
+            "create_display_platform", MIR_SERVER_GRAPHICS_PLATFORM_VERSION);
```

Comment:
"create_display_platform" is unused by Mir. This should be "create_host_platform" as used in src/server/graphics/default_configuration.cpp.

Reply:
I'm not sure about this (and, likewise, the rendering platform one).

I was testing the Platform API we *want* to have, rather than the platform API we currently have, and in practice at the moment `create_display_platform` just returns an aggregate `DisplayPlatform` and `RenderingPlatform`…

Reply:
While I get the motivation for testing what it should be, that isn't an indicator of what is needed for Mir to work. Shouldn't the test reflect the current requirement and change as that requirement is refined?